src/routes/news/containers/WithCustomNewsProvider.js

```diff
@@ -0,0 +1,357 @@
+        const { selectedNewsType } = this.state
+        return (
+          <HeaderContainer>
+            {newsTabs.map(tab =>
+              cityModel[tab.toggleAttr] ? (
+                <TranslatedNewsTypeItem
+                  key={tab.type}
+                  tab={tab}
+                  selectedNewsType={selectedNewsType}
+                  onItemPress={this.selectNewsItemAndScrollToTop}
+                />
+              ) : null
+            )}
+          </HeaderContainer>
+        )
+      };
+
+      refresh = () => {
+        const props = this.props
+        if (
+          props.status === 'routeNotInitialized' ||
+          props.status === 'loading' ||
+          props.status === 'languageNotAvailable'
+        ) {
+          throw Error(
+            'Refreshing is not possible because the route is not yet initialized or already loading.'
+          )
+        }
+
+        if (props.refreshProps) {
+          refresh(props.refreshProps, props.dispatch)
+        }
+      };
+
+      changeUnavailableLanguage = (newLanguage: string) => {
+        if (this.props.status !== 'languageNotAvailable') {
+          throw Error(
+            'Call of changeUnavailableLanguage is only possible when language is not available.'
+          )
+        }
+        this.props.changeUnavailableLanguage(this.props.dispatch, newLanguage)
+      };
+
+      render () {
+        const props = this.props
+
+        if (props.status === 'routeNotInitialized') {
+          return null
+        } else if (props.status === 'error') {
+          return (
+            <ScrollView
+              refreshControl={
+                <RefreshControl onRefresh={this.refresh} refreshing={false} />
+              }
+              contentContainerStyle={{ flexGrow: 1 }}>
+              <FailureContainer
+                tryAgain={this.refresh}
+                message={props.message}
+                code={props.code}
+              />
+            </ScrollView>
+          )
+        } else if (props.status === 'languageNotAvailable') {
+          return (
+            <LanguageNotAvailableContainer
+              languages={props.availableLanguages}
+              changeLanguage={this.changeUnavailableLanguage}
+            />
+          )
+        } else if (props.status === 'loading') {
+          return (
+            <View style={{ flex: 1 }}>
+              {!props.innerProps?.path &&
+                props.innerProps?.cityModel &&
+                this.renderHeader(props.innerProps?.cityModel)}
+              <Loader />
+            </View>
+          )
+        } else {
+          return (
+            <View style={{ flex: 1 }}>
+              {!props.innerProps.path &&
+                this.renderHeader(props.innerProps.cityModel)}
+              <Component {...props.innerProps} dispatch={props.dispatch} />
+            </View>
+          )
+        }
+      }
+    }
+  }
+}
+
+export default withCustomNewsProvider
```

Comment:
Way to big component with too much fucntionality. Same for some other components. Please try to split.

Reply:
Tried to clean than splitting it. bc, it didn't sound ok to me to split it. If you have any idea please let me know.